pkg/platform/kube/functionres/lazy.go

```diff
@@ -332,15 +332,6 @@ func (lc *lazyClient) WaitAvailable(ctx context.Context,
 				err, functionState := lc.waitFunctionDeploymentReadiness(ctx,
 					function,
 					functionResourcesCreateOrUpdateTimestamp)
-				if err == nil {
```

Comment:
look on line 341 (after your change), `"err", err.Error(),`
it assumes the `err` is not nill. if you removed the `err == nil` check above it, you are exposed to a panic. perhaps you want to validate that?